### DISCUSSION TITLE
Trouble upgrading from deck.gl 7(.3.15) => deck.gl 8(latest)

### DISCUSSION THREAD

**Discussion by mcchong** [visgl/deck.gl#5253] [Q&A] [answered]
Hi,

I've been trying to update from deck.gl 7 -> 8, but for some reason my layers arent' showing.  I'm still passing them the same data, and am getting logging that they are rendering, but am not seeing the layers visible on the map. I'm using deck.gl react with a `StaticMap` from react-map-gl, and some geojson/tilelayers.  Strangely, I've tried creating a totally new component with code out of a basic example, and it's still not visible.

I have a hunch maybe there's a peer dependency or something that's also out of date?  Working on trying to create a minimal reproduction, but in the meantime has anyone else seen this issue?  I've been digging through the changelog and upgrade notes, but nothing I've tried seems to do the trick.

Some unrelated background is that I was having some issues on deckgl 7 with layers randomly becoming out of sync (I think the zoom was off somehow), when updating the viewstate in the way that I was (e.g. panning/zooming to a bbox based on the user selecting a set of points).  I tried reproducing it in a sandbox, couldn't using the latest version of deck.gl, and then couldn't get things to compile in the sandbox using the same version of deck.gl here: https://codesandbox.io/s/cranky-cache-ts4v9

**Comment by Pessimistress** [comment, marked as answer]
Your sandbox loads fine for me. It may encounter issues if you change dependency versions and then hot reload, but it does figure it out if you refresh the page.

**Reply by mcchong** [reply]
Sorry, just trying to create a reproduction with the package versions I'm currently using, and I can't get it to compile -> maybe it's a codesandbox issue, but if you have the chance, any idea what's happening here?  I'm a bit stumped:
https://codesandbox.io/s/proud-hooks-rm2ln

**Reply by mcchong** [reply]
Ah, you have to mouseover the map to trigger the npe.  It's pretty much just the example from https://deck.gl/docs/get-started/using-with-react, with the package versions I'm currently running...

**Reply by Pessimistress** [reply]
Your sandbox works with the latest deck.gl release. It currently uses 7.3.15 and does not have a compatible version of viewport-mercator-project installed.
